model/src/main/scala/za/co/absa/atum/model/AdditionalData.scala

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright 2021 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.atum.model
+
+case class AdditionalData(
+    additionalData: Map[String, String]
+)
```

Comment:
Do we need the class if it has just one field? We could use the field directly. The same thing with partitioning.

Reply:
I was thinking the same. I think that potentially if we decide to extend them, which might be highly likely, then it makes sense to keep it as is.

In any case, especially with Partitioning, I wanted to keep the information about the overall partitioning structure close and encapsulated, i.e. as a separated class. If I keep only Partition class and move the `Seq[Partition]` into `Checkpoint`, then I have to implement `Seq[Partition]` on other places just to enforce the order of partitions. By having it like this, everyone can use Partitioning with pre-described overall structure, i.e. saying that Partitions are related to each other in Seq way and order is important.

Not sure if my explanation makes sense though :D ?

@benedeki do you have an opinion about this? What about some sort of partition / metadata versioning? Or some other attributes?

Reply:
I think at the beginning we can omit these very simple types. (This and for example `Partition`)